ledger_device_sdk/src/nbgl.rs

```diff
@@ -371,6 +375,446 @@ impl<'a> NbglReview<'a> {
+            icon: icon.map_or(None, |g| Some(g.into())),
+            on_top: on_top,
+            style: style,
+            offset_y: offset_y,
+        }
+    }
+}
+
+/// Structure exposed by the NBGL Rust API to the user to create a
+/// "long press" button to confirm some information that will be displayed
+/// on the device when using the NbglGenericReview struct.
+pub struct InfoLongPress {
+    text: CString,
+    icon: Option<nbgl_icon_details_t>,
+    long_press_text: CString,
+    tune_id: TuneIndex,
+}
+
+impl InfoLongPress {
+    pub fn new(
+        text: &str,
+        icon: Option<&NbglGlyph>,
+        long_press_text: &str,
+        tune_id: TuneIndex,
+    ) -> InfoLongPress {
+        InfoLongPress {
+            text: CString::new(text).unwrap(),
+            icon: icon.map_or(None, |g| Some(g.into())),
+            long_press_text: CString::new(long_press_text).unwrap(),
+            tune_id: tune_id,
+        }
+    }
+}
+
+/// Structure exposed by the NBGL Rust API to the user to create a
+/// button to confirm some information that will be displayed
+/// on the device when using the NbglGenericReview struct.
+pub struct InfoButton {
+    text: CString,
+    icon: Option<nbgl_icon_details_t>,
+    button_text: CString,
+    tune_id: TuneIndex,
+}
+
+impl InfoButton {
+    pub fn new(
+        text: &str,
+        icon: Option<&NbglGlyph>,
+        button_text: &str,
+        tune_id: TuneIndex,
+    ) -> InfoButton {
+        InfoButton {
+            text: CString::new(text).unwrap(),
+            icon: icon.map_or(None, |g| Some(g.into())),
+            button_text: CString::new(button_text).unwrap(),
+            tune_id: tune_id,
+        }
+    }
+}
+
+/// Structure exposed by the NBGL Rust API to the user to create a
+/// tag/value list screen that will be displayed on the device when
+/// using the NbglGenericReview struct.
+pub struct TagValueList {
+    pairs: Vec<nbgl_contentTagValue_t>,
+    items: Vec<CString>,
+    values: Vec<CString>,
+    nb_max_lines_for_value: u8,
+    small_case_for_value: bool,
+    wrapping: bool,
+}
+
+impl TagValueList {
+    pub fn new(
+        pairs: &[Field],
+        nb_max_lines_for_value: u8,
+        small_case_for_value: bool,
+        wrapping: bool,
+    ) -> TagValueList {
+        let mut c_field_strings: Vec<nbgl_contentTagValue_t> = Vec::new();
+        let mut c_field_names: Vec<CString> = Vec::new();
+        let mut c_field_values: Vec<CString> = Vec::new();
+        for field in pairs {
+            let name = CString::new(field.name).unwrap();
+            let value = CString::new(field.value).unwrap();
+            let tag_value = nbgl_contentTagValue_t {
+                item: name.as_ptr() as *const c_char,
+                value: value.as_ptr() as *const c_char,
+                ..Default::default()
+            };
+            c_field_strings.push(tag_value);
+            c_field_names.push(name);
+            c_field_values.push(value);
+        }
```

Comment:
As we know in advance the number of items that `c_field_strings`, `c_field_names` and `c_field_values` will have, it is better to initialize the vectors using [Vec::with_capacity](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.with_capacity) . This will reduce the number of allocations/frees made in the heap.

Reply:
... or you can use `collect` for each vector, which should be optimized.

Reply:
I have used `with_capacity`